### ISSUE TITLE
Null pointer exception crashes game

### ISSUE THREAD

**Issue by Thimoteus** [00-Evan/shattered-pixel-dungeon-gdx#14]
```
Exception in thread "Thread-1" java.lang.NullPointerException
	at com.watabou.noosa.BitmapText.measure(BitmapText.java:174)
	at com.shatteredpixel.shatteredpixeldungeon.ui.UpdateNotification.layout(UpdateNotification.java:141)
	at com.shatteredpixel.shatteredpixeldungeon.ui.UpdateNotification.updateMessage(UpdateNotification.java:173)
	at com.shatteredpixel.shatteredpixeldungeon.ui.UpdateNotification.access$200(UpdateNotification.java:41)
	at com.shatteredpixel.shatteredpixeldungeon.ui.UpdateNotification$1.run(UpdateNotification.java:101)
Exception in thread "Thread-2" java.lang.NullPointerException
	at com.watabou.noosa.BitmapText.measure(BitmapText.java:174)
	at com.shatteredpixel.shatteredpixeldungeon.ui.UpdateNotification.layout(UpdateNotification.java:141)
	at com.shatteredpixel.shatteredpixeldungeon.ui.UpdateNotification.updateMessage(UpdateNotification.java:173)
	at com.shatteredpixel.shatteredpixeldungeon.ui.UpdateNotification.access$200(UpdateNotification.java:41)
	at com.shatteredpixel.shatteredpixeldungeon.ui.UpdateNotification$1.run(UpdateNotification.java:101)
GAME: Welcome to the level 11 of Pixel Dungeon!
GAME: A hidden pressure plate clicks!
Exception in thread "LWJGL Application" java.lang.NullPointerException
	at com.shatteredpixel.shatteredpixeldungeon.levels.traps.Trap.reveal(Trap.java:57)
	at com.shatteredpixel.shatteredpixeldungeon.levels.traps.Trap.trigger(Trap.java:75)
	at com.shatteredpixel.shatteredpixeldungeon.levels.Level.press(Level.java:841)
	at com.shatteredpixel.shatteredpixeldungeon.actors.hero.Hero.move(Hero.java:1363)
	at com.shatteredpixel.shatteredpixeldungeon.actors.hero.Hero.getCloser(Hero.java:1031)
	at com.shatteredpixel.shatteredpixeldungeon.actors.hero.Hero.actMove(Hero.java:538)
	at com.shatteredpixel.shatteredpixeldungeon.actors.hero.Hero.act(Hero.java:453)
	at com.shatteredpixel.shatteredpixeldungeon.actors.hero.Hero.handle(Hero.java:1100)
	at com.shatteredpixel.shatteredpixeldungeon.scenes.GameScene$2.onSelect(GameScene.java:792)
	at com.shatteredpixel.shatteredpixeldungeon.scenes.CellSelector.select(CellSelector.java:176)
	at com.shatteredpixel.shatteredpixeldungeon.scenes.CellSelector.onClick(CellSelector.java:64)
	at com.watabou.noosa.TouchArea.onSignal(TouchArea.java:121)
	at com.watabou.noosa.TouchArea.onSignal(TouchArea.java:23)
	at com.watabou.utils.Signal.dispatch(Signal.java:76)
	at com.watabou.pd.desktop.DesktopInputProcessor.touchUp(DesktopInputProcessor.java:143)
	at com.badlogic.gdx.backends.lwjgl.LwjglInput.processEvents(LwjglInput.java:326)
	at com.badlogic.gdx.backends.lwjgl.LwjglApplication.mainLoop(LwjglApplication.java:199)
```
[Here's](http://imgur.com/sjobEro,zcCE8t2#1) the screen right before it crashes.

I click on the tile just to the left of the leftmost visible tile and the game closes.

I've also saved all the .dat files and can upload them too if you need.

EDIT: I think [this](http://imgur.com/zSTiLH8) might be related? I used my wand of prismatic light and revealed the pitfall trap where the game would have taken me automatically.

**Comment by 00-Evan** [comment]
Yup, this is due to some specific ways that pitfall traps handle their sprite, causes a crash in certain situations. (the UpdateNotification is unrelated and doesn't crash, but I should probably fix that regardless).

I'm currently working on 0.3.1d, which will fix this issue along with others, should be out in a few days.

**Comment by 00-Evan** [comment]
This was fixed in the latest update(0.3.1d)
